packages/grafana-ui/src/components/PageLayout/PageToolbar.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { FC, ReactNode } from 'react';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { useStyles } from '../../themes/ThemeContext';
+import { IconName } from '../../types';
+import { Icon } from '../Icon/Icon';
+import { styleMixins } from '../../themes';
+import { IconButton } from '../IconButton/IconButton';
+import { selectors } from '@grafana/e2e-selectors';
+
+export interface Props {
+  pageIcon?: IconName;
+  title: string;
+  parent?: string;
+  onGoBack?: () => void;
+  onClickTitle?: () => void;
+  onClickParent?: () => void;
+  leftItems?: ReactNode[];
+  children?: ReactNode;
+  className?: string;
+  isFullscreen?: boolean;
+}
+
+/** @alpha */
+export const PageToolbar: FC<Props> = React.memo(
+  ({
+    title,
+    parent,
+    pageIcon,
+    onGoBack,
+    children,
+    onClickTitle,
+    onClickParent,
+    leftItems,
+    isFullscreen,
+    className,
+  }) => {
+    const styles = useStyles(getStyles);
+
+    /**
+     * .page-toolbar css class is used for some legacy css view modes (TV/Kiosk) and
+     * media queries for mobile view when toolbar needs left padding to make room
+     * for mobile menu icon. This logic hopefylly can be changed when we move to a full react
+     * app and change how the app side menu & mobile menu is rendered.
+     */
+    const mainStyle = cx(
+      'page-toolbar',
+      styles.toolbar,
+      {
+        ['page-toolbar--fullscreen']: isFullscreen,
+      },
+      className
+    );
+
+    return (
+      <div className={mainStyle}>
+        <div className={styles.toolbarLeft}>
+          {pageIcon && !onGoBack && (
+            <div className={styles.pageIcon}>
+              <Icon name={pageIcon} size="lg" />
+            </div>
+          )}
+          {onGoBack && (
+            <div className={styles.goBackButton}>
+              <IconButton
+                name="arrow-left"
+                tooltip="Go back (Esc)"
+                tooltipPlacement="bottom"
+                size="xxl"
+                surface="dashboard"
+                aria-label={selectors.components.BackButton.backArrow}
+                onClick={onGoBack}
+              />
+            </div>
+          )}
+          <div className={styles.titleWrapper}>
+            {parent && onClickParent && (
+              <a onClick={onClickParent} className={cx(styles.titleLink, styles.parentLink)}>
```

Comment:
This should be either a link with ref or a button with onClick.

Reply:
yes, was copied from DashNav this way but your right this should be a button